client/src/utils/utils.test.js

```diff
@@ -0,0 +1,17 @@
+import Utils from "./utils";
+
+describe("test utils", () => {
+    it("test isEmpty", async () => {
+        expect(Utils.isEmpty([])).toBe(true); //  "Empty array");
+        expect(Utils.isEmpty(["data", undefined])).toBe(true); // , "Array contains `undefined`");
+        expect(Utils.isEmpty(["data", null])).toBe(true); //, "Array contains `null`");
+        expect(Utils.isEmpty(["data", "__null__"])).toBe(true); // , "Array contains `__null__`");
+        expect(Utils.isEmpty(["data", "__undefined__"])).toBe(true); // , "Array contains `__undefined__`");
```

Comment:
Offtopic: maybe this is some JS well-known truth, but it deeply confuses me... those arrays a definitely NOT empty they have "data". I would expect false in those cases, and maybe other utilities like `hasNull` or `hasUndefined`?
I was just curious, nothing that needs to be addressed here.

Reply:
Yeah, this one got me when I read through it.  We could probably rename this utility function something better, or, maybe work towards just eliminating this kind of logic -- it's a Galaxy-ism, saying it's empty if it doesn't have any unselectable values I think.  Those arrays are definitely *not* empty :)